transport/dependencies.go

```diff
@@ -20,17 +20,19 @@
 
 package transport
 
+import "github.com/opentracing/opentracing-go"
+
 // Deps is the interface of any object useful for passing injected
 // dependencies into inbound and outbound transports.
 type Deps interface {
-	// Tracer() opentracing.Tracer
+	Tracer() opentracing.Tracer
 }
 
 // NoDeps is a no-op implementation of Deps
 var NoDeps noDeps
 
 type noDeps struct{}
 
-// func (d noDeps) Tracer() opentracing.Tracer {
-// 	return opentracing.NoopTracer{}
-// }
+func (d noDeps) Tracer() opentracing.Tracer {
+	return opentracing.NoopTracer{}
```

Comment:
fwiw, in tchannel-go I've implemented a similar method on TChannel that either returns a Tracer instance it was initialized with, or returns `opentracing.GlobalTracer()`, which by default happens to return a singleton instance of `NoopTracer`. In Go the use of global variables is not frowned upon as say in Java, so this pattern allows minimal work to enable tracing in a service, one just needs to store a real tracer in the global variable, e.g. via `opentracing.InitGlobalTracer(someTracerImpl)`, while the instances of TChannel do not need to be explicitly given any tracer since they will default to that global.